Chore/release process improvements (#130)
americanexpress/one-app: RELEASE.md

## Automated release process

>  1. An automated pull request will be raised every Wednesday at 16:00 UTC, from a `prepare-release` branch to master. This uses [pull_request_release workflow](.github/workflows/pull_request_release.yml) and updates the [package.json](package.json), [package-lock.json](package-lock.json), [one-app-statics package.json](one-app-statics/package.json) and runs `npm run release:changelog` to generate the changelog. Behind the scene it uses [standard-version](https://github.com/conventional-changelog/standard-version) to generate and update these files with the changes for released. The commit message contains will be in the following format `chore(release): X.X.X` . You can update this pull request to remove or add any new changes.

Reviewer: ```suggestion
 1. An automated pull request will be raised every Wednesday at 16:00 UTC, from a `prepare-release` branch to master. This uses [pull_request_release workflow](.github/workflows/pull_request_release.yml) and updates the [package.json](package.json), [package-lock.json](package-lock.json), [one-app-statics package.json](one-app-statics/package.json) and runs `npm run release:changelog` to generate the changelog. Behind the scenes it uses [standard-version](https://github.com/conventional-changelog/standard-version) to generate and update these files with the changes to be released. The commit message will be in the following format `chore(release): X.X.X` . You can update this pull request to add or remove any new changes.
```

5. Update the draft release notes with any changes required and tie it to the released tag.
6. For the release notes use the below format. Please review some of the releases to check the format used [https://github.com/americanexpress/one-app/releases](https://github.com/americanexpress/one-app/releases)

  ```
      [Paste changelog entries here]
    
      ### Docker Images

      #### Developer Image
      [one-app-dev/version-released](https://hub.docker.com/layers/oneamex/one-app-dev/link-to-the-docker-tag)

      #### Production Image
      [one-app/version-released](https://hub.docker.com/layers/oneamex/one-app/link-to-the-docker-tag)

      ### One App Statics
      https://www.npmjs.com/package/@americanexpress/one-app-statics

  ```
7. 

## Manual release process

This process can be used to make ad hoc releases outside of wednesday release cycle.

 1. Run `npm run release` locally within your branch, this would update [package.json](package.json), [package-lock.json](package-lock.json)and [one-app-statics package.json](one-app-statics/package.json) with the new version to be released. 
 2. Run `npm run release:changelog` to generate and update the changelog.

>  3. Push your changes and create a pull request to master.

Reviewer 2: ```suggestion
 3. Commit using the `chore(release): X.X.X` message format and push your changes
 and create a pull request to master.
```

Reviewer 2: Would this need to be on a specific branch?

Author: nope, although for consistency using the `prepare-release` branch would be a better suggestions. I'll mention that

   > Integration tests will continue to run in Travis as they currently do, after the automated pull request is created it would run the tests on `prepare-release` branch.

1. The merge will trigger the automatic generation of a new tag using the semantic version provided during the merging of the pull request above.
2. After the the generated tag is pushed to the branch this will trigger the docker build and publish the statics and push the images to Docker Hub. The development and production images would be accessible in docker [https://hub.docker.com/u/oneamex](https://hub.docker.com/u/oneamex)

> 3. The statics assets will be published and added to a tag that has been released via the [release static assets action](.github/workflows/docker_release.yml)

Reviewer 3: ```suggestion
3. The statics assets will be published and added to the tag created via the [release static assets action](.github/workflows/docker_release.yml) as part of the previous step.
```

> 4. We are currently using [https://github.com/release-drafter/release-drafter](https://github.com/release-drafter/release-drafter) to generate release notes. This is a github action that generates draft releases note that can be added to the released tag. Please add the labels specified within [release-drafter](.github/release-drafter.yml) to categorize different pull requests to ensure that any changes are categorized correctly. You can also use the changelog generated for this content.

Reviewer 3: does release-drafter have to use labels or can it use commits ? can we auto label based of the commit.
If we take the changes add as part of the changelog do we even need release drafter ?

Author: We have some work to address this in a separate PR

> 5. Update the draft release notes with any changes required and tie it to the released tag.

Reviewer 3: what do you mean by "tie it" ? 
```suggestion
5. Update the draft release notes with any changes required and edit the released tag to add them there.
```

6. For the release notes use the below format. Please review some of the releases to check the format used [https://github.com/americanexpress/one-app/releases](https://github.com/americanexpress/one-app/releases)

  ```

>       [Paste changelog entries here]

Reviewer 3: is all this white space necessary ?

Author: Remove this, i had missed this earlier

2. After the the generated tag is pushed to the branch this will trigger the docker build and publish the statics and push the images to Docker Hub. The development and production images would be accessible in docker [https://hub.docker.com/u/oneamex](https://hub.docker.com/u/oneamex)

3. We are currently using [https://github.com/release-drafter/release-drafter](https://github.com/release-drafter/release-drafter) to generate release notes. Please add the labels specified within [release-drafter](.github/release-drafter.yml) to categorize different pull requests. Update the draft release notes and tie it to the release tag above, you can also link this to different artifacts. The statics assets will be published and added to a tag that has been released.

> 4. For the release notes use the below format, checkout [https://github.com/americanexpress/one-app/releases](https://github.com/americanexpress/one-app/releases)

Reviewer 3: what do you mean by "checkout" ?
It's not very clear what is meant to happen with the release notes. where do they come from ? how do you create them ? what is the complete process

Author: Updated this to be more clear, let me know what you think. Provided a template for the release notes below this.

   
   ```

> 

Reviewer 3: remove this whitespace

Author: Updated 0415f55

### How can I run the first release?

For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --first-release` to update the package.json files.

### How can I create a prerelease?

> For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --prerelease` to update the package.json files.

Reviewer 2: ```suggestion
For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --prerelease` to update the package.json files.

For dry running the output from release, you can run `npm run release -- --prerelease --dry-run`.
```

> 

Reviewer 2: ```suggestion

For dry running the output from release, you can run `npm run release -- --prerelease --dry-run`.
```
